### ISSUE TITLE
Add tests around checkpoint loading

### ISSUE THREAD

**Issue by suchenzang** [facebookresearch/metaseq#268]
## 🚀 Feature Request
Add tests around loading checkpoints!

### Motivation

Right now we occasionally break codepaths related to checkpoint loading, since it isn't frequently tested internally until someone has to... load a checkpoint.  We will need to start adding some tests around this, since this area will also go through some refactoring to address https://github.com/facebookresearch/metaseq/issues/78

**Comment by punitkoura** [comment]
We had added this test some time back to test checkpoint loading for a 125m model, and ensuring that the logits generated by the model parallel version and singleton version are the same.

We can make this more thorough to catch other errors as well...

**Comment by suchenzang** [comment]
Another related issue: checkpoint downloading (this is at the intersection of two brittle components: checkpointing + our logic around different compute envs). https://github.com/facebookresearch/metaseq/pull/272
